ctapipe_io_lst/event_time.py

```diff
@@ -33,26 +31,39 @@ def validate(self, obj, value):
         return super().validate(obj, value)
 
 
-def calc_dragon_time(lst_event_container, module_index, reference_time, reference_counter):
+def calc_dragon_time(pps_counter, tenMHz_counter, reference_time, reference_counter):
     '''
     Calculate a unix tai timestamp (in ns) from dragon counter values
     and reference time / counter value for a given module index.
     '''
-    pps_counter = lst_event_container.evt.pps_counter[module_index]
-    tenMHz_counter = lst_event_container.evt.tenMHz_counter[module_index]
-    return (
-        reference_time
-        + combine_counters(pps_counter, tenMHz_counter)
-        - reference_counter
-    )
+    reference_time = reference_time.astype(np.uint64)
+    reference_counter = reference_counter.astype(np.uint64)
+
+    counter = combine_counters(pps_counter, tenMHz_counter)
+
+    if counter > reference_counter:
+        delta = counter - reference_counter
+        timestamp = reference_time + delta
+    else:
+        delta = reference_counter - counter
+        timestamp = reference_time - delta
+
+    return timestamp
+
+
+S_TO_NS = np.uint64(1e9)
+TEN_MHZ_TO_NS = np.uint64(100)
 
 
 def combine_counters(pps_counter, tenMHz_counter):
     '''
     Combines the values of pps counter and tenMHz_counter
     and returns the sum in ns.
     '''
-    return int(1e9) * pps_counter + 100 * tenMHz_counter
+    return (
+        S_TO_NS * pps_counter.astype(np.uint64)
```

Comment:
instead of using variables like `S_TO_NS` or `TEN_MHZ_TO_NS` could it be a good idea to use astropy units or may that complicate things innecessarily?

Reply:
astropy units by default convert everything to float64. We really need uint64 values here. 

The astropy units code would be much more verbose to ensure we have the right dtypes and I don't think it would really benefit us here.

Reply:
good, I also think it should be ok to carry on as it is now